db/migration/1756391135333-AddSessionUserForeignKey.ts

```diff
@@ -0,0 +1,43 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddSessionUserForeignKey1756391135333
+    implements MigrationInterface
+{
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
+            ALTER TABLE sessions
+            ADD COLUMN user_id INT AFTER session_key,
+            ADD FOREIGN KEY (user_id) REFERENCES users(id)
+            ON DELETE CASCADE ON UPDATE RESTRICT
+        `)
+
+        // The substring_index calls extract the email address out of the base64-encoded
+        // and concatenated session_data string, such that we can join on the users table
+        await queryRunner.query(`
+            WITH sessions_with_emails AS (
+                SELECT session_key,
+                SUBSTRING_INDEX(
+                    SUBSTRING_INDEX(from_base64(session_data), '":"', -1),
+                    '"}',
+                    1) AS email
+                FROM sessions)
```

Comment:
It probably would've been fine to drop all existing sessions as part of this migration, forcing all users to re-login when this lands.

But decoding the `session_data` field in MySQL is actually not too tough, so I opted to go with that instead. Let me know if you disagree.

Reply:
Haha, it's simply the Google Auth screen that sometimes pops up. Nothing major.